p2p/dandelion/src/txpool.rs

```diff
@@ -0,0 +1,545 @@
+                    self.store_tx_and_stem(tx, tx_id, Some(from)).await
+                }
+            }
+            TxState::Fluff => {
+                tracing::debug!("Fluffing incoming tx");
+                self.store_and_fluff_tx(tx, tx_id).await
+            }
+            TxState::Local => {
+                // If we have already stemed this tx then nothing to do.
+                if have_tx.is_some() {
+                    tracing::debug!("Received a local tx that we already have, skipping");
+                    return Ok(());
+                }
+                tracing::debug!("Steming local transaction");
+                self.store_tx_and_stem(tx, tx_id, None).await
+            }
+        }
+    }
+
+    /// Promotes a tx to the clear pool.
+    async fn promote_tx(&mut self, tx_id: TxID) -> Result<(), tower::BoxError> {
+        // Remove the tx from the maps used during the stem phase.
+        self.stem_origins.remove(&tx_id);
+
+        // The key for this is *Not* the tx_id, it is given on insert, so just keep the timer in the
+        // map. These timers should be relatively short, so it shouldn't be a problem.
+        //self.embargo_timers.try_remove(&tx_id);
+
+        self.backing_pool
+            .ready()
+            .await?
+            .call(TxStoreRequest::Promote(tx_id))
+            .await?;
+
+        Ok(())
+    }
+
+    /// Promotes a tx to the public fluff pool and fluffs the tx.
+    async fn promote_and_fluff_tx(&mut self, tx_id: TxID) -> Result<(), tower::BoxError> {
+        tracing::debug!("Promoting transaction to public pool and fluffing it.");
+
+        let TxStoreResponse::Transaction(tx) = self
+            .backing_pool
+            .ready()
+            .await?
+            .call(TxStoreRequest::Get(tx_id.clone()))
+            .await?
+        else {
+            panic!("Backing tx pool responded with wrong response for request.");
+        };
+
+        let Some((tx, state)) = tx else {
+            tracing::debug!("Could not find tx, skipping.");
+            return Ok(());
+        };
+
+        if state == State::Fluff {
+            tracing::debug!("Transaction already fluffed, skipping.");
+            return Ok(());
+        }
+
+        self.promote_tx(tx_id.clone()).await?;
+        self.fluff_tx(tx, tx_id).await
+    }
+
+    /// Returns a tx stored in the fluff _OR_ stem pool.
+    async fn get_tx_from_pool(&mut self, tx_id: TxID) -> Result<Option<Tx>, tower::BoxError> {
+        let TxStoreResponse::Transaction(tx) = self
+            .backing_pool
+            .ready()
+            .await?
+            .call(TxStoreRequest::Get(tx_id))
+            .await?
+        else {
+            panic!("Backing tx pool responded with wrong response for request.");
+        };
+
+        Ok(tx.map(|tx| tx.0))
+    }
+
+    async fn run(
+        mut self,
+        mut rx: mpsc::Receiver<(DandelionPoolRequest<Tx, TxID, PID>, oneshot::Sender<()>)>,
+    ) {
```

Comment:
This starts the txpool?

Reply:
It's the pools task, so this is function you put in a `tokio::spawn`